### PR TITLE
refactor backend

### PR DESCRIPTION
**Pull request by Rampage1xx**
Hello, 

This PR aims to improve golang backend code without mutating the folder structure.

### CODE REVIEW

File: internal/kubeconfig/serviceaccount.go [hunk 1]
```diff
@@ -28,15 +28,16 @@ func getServiceAccountToken(ctx context.Context, c clientset.Interface, name str
 		log.Printf("Service Account not created: %v", err)
 	}
 
-	// Get token
-	_, token, err = getReferencedServiceAccountToken(c.(*clientset.Clientset), ns, name, true)
+	// check for errors
+	_, _, err = getReferencedServiceAccountToken(c.(*clientset.Clientset), ns, name, true)
 	if err != nil {
 		log.Fatal(err)
 	}
```
Comment (phisco) [inline internal/kubeconfig/serviceaccount.go]: looks like you are not returning the token anymore, is this intentional?

File: internal/server/handlers.go [hunk 2]
```diff
@@ -136,18 +136,18 @@ func CreateClusterRole(c echo.Context) error {
...
+	_, err := ac.Kubeclient.RbacV1().ClusterRoles().Create(c.Request().Context(), &rbacv1.ClusterRole{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: r.RoleName,
 		},
 		Rules: r.Rules,
 	}, metav1.CreateOptions{})
 
-	return c.JSON(http.StatusOK, Response{Ok: true})
+	if err != nil {
+		panic(err.Error())
```
Comment (phisco) [inline internal/server/handlers.go:147]: why not log.Fatalf?

File: internal/server/handlers.go [hunk 3]
```diff
@@ -186,7 +182,11 @@ func CreateRolebinding(c echo.Context) error {
 		Subjects: r.Subjects,
 	}, metav1.CreateOptions{})
 
-	return c.JSON(http.StatusOK, Response{Ok: true})
+	if err != nil {
+		panic(err.Error())
```
Comment (phisco) [inline internal/server/handlers.go:186]: same as above

File: internal/server/handlers.go [hunk 4]
```diff
@@ -239,18 +232,20 @@ func deleteClusterRole(c echo.Context) error {
...
 	}
 
-	ac.Kubeclient.RbacV1().ClusterRoles().Delete(c.Request().Context(), r.RoleName, metav1.DeleteOptions{})
-	return c.JSON(http.StatusOK, Response{Ok: true})
+	err := ac.Kubeclient.RbacV1().ClusterRoles().Delete(c.Request().Context(), r.RoleName, metav1.DeleteOptions{})
+
+	if err != nil {
+		panic(err.Error())
+	}
+
```
Comment (phisco) [inline internal/server/handlers.go:240]: same as above

File: internal/server/handlers.go [hunk 5]
```diff
@@ -260,8 +255,13 @@ func deleteClusterRolebinding(c echo.Context) error {
...
 	}
 
-	ac.Kubeclient.RbacV1().ClusterRoleBindings().Delete(c.Request().Context(), r.RolebindingName, metav1.DeleteOptions{})
-	return c.JSON(http.StatusOK, Response{Ok: true})
+	err := ac.Kubeclient.RbacV1().ClusterRoleBindings().Delete(c.Request().Context(), r.RolebindingName, metav1.DeleteOptions{})
+
+	if err != nil {
+		panic(err.Error())
+	}
+
```
Comment (phisco) [inline internal/server/handlers.go:263]: same as above

File: internal/server/handlers.go [hunk 6]
```diff
@@ -283,8 +280,13 @@ func deleteRole(c echo.Context) error {
 		return c.JSON(http.StatusBadRequest, ErrorRes{err.Error()})
 	}
 
-	ac.Kubeclient.RbacV1().Roles(r.Namespace).Delete(c.Request().Context(), r.RoleName, metav1.DeleteOptions{})
-	return c.JSON(http.StatusOK, Response{Ok: true})
+	err := ac.Kubeclient.RbacV1().Roles(r.Namespace).Delete(c.Request().Context(), r.RoleName, metav1.DeleteOptions{})
+
+	if err != nil {
+		panic(err.Error())
+	}
```
Comment (phisco) [inline internal/server/handlers.go:287]: same as above

File: internal/server/handlers.go [hunk 7]
```diff
@@ -305,8 +304,13 @@ func deleteRolebinding(c echo.Context) error {
 		return c.JSON(http.StatusBadRequest, ErrorRes{err.Error()})
 	}
 
-	ac.Kubeclient.RbacV1().RoleBindings(r.Namespace).Delete(c.Request().Context(), r.RolebindingName, metav1.DeleteOptions{})
-	return c.JSON(http.StatusOK, Response{Ok: true})
+	err := ac.Kubeclient.RbacV1().RoleBindings(r.Namespace).Delete(c.Request().Context(), r.RolebindingName, metav1.DeleteOptions{})
+
+	if err != nil {
+		panic(err.Error())
+	}
```
Comment (phisco) [inline internal/server/handlers.go:311]: same as above

**Review by phisco — CHANGES_REQUESTED** [review]
minor fixes

File: internal/server/handlers.go (hunk 2 above)
Comment (Rampage1xx) [reply internal/server/handlers.go:147]: Uniformity. Panic has been used in the rest of the file so im handling the error in the say way, we can decide to switch all panic to fatal

File: internal/kubeconfig/serviceaccount.go (hunk 1 above)
Comment (Rampage1xx) [reply internal/kubeconfig/serviceaccount.go]: the comment branch is outdated I think, i see the token in my version in an explicit way

Comment (phisco) [reply internal/kubeconfig/serviceaccount.go]: right, my fault

**Comment by angelbarrera92** [comment]
@Rampage1xx  Branch need a rebase. 
@nandajavarma can you review this PR?

**Review by lzecca78 — APPROVED** [review]
LGTM ❤️

**Comment by angelbarrera92** [comment]
How confident we are to push this PR to master? Can we rebase it? Can we have a quick demo around the refactor?

**Comment by Rampage1xx** [comment]
Hello,

In my opinion, without extensive (manual) tests it shouldn't be merged. 

Happy to give a demo if needed around the code refactoring

**Comment by ipedrazas** [comment]
If extensive manual testing is required, the PR is not ready.

**Pull request merged**